exp/controllers/awsmanagedmachinepool_controller.go

```diff
@@ -192,15 +226,40 @@ func (r *AWSManagedMachinePoolReconciler) reconcileNormal(
 func (r *AWSManagedMachinePoolReconciler) reconcileDelete(
 	_ context.Context,
 	machinePoolScope *scope.ManagedMachinePoolScope,
+	ec2Scope scope.EC2Scope,
 ) (ctrl.Result, error) {
 	machinePoolScope.Info("Reconciling deletion of AWSManagedMachinePool")
 
 	ekssvc := eks.NewNodegroupService(machinePoolScope)
+	ec2Svc := ec2.NewService(ec2Scope)
 
 	if err := ekssvc.ReconcilePoolDelete(); err != nil {
 		return reconcile.Result{}, errors.Wrapf(err, "failed to reconcile machine pool deletion for AWSManagedMachinePool %s/%s", machinePoolScope.ManagedMachinePool.Namespace, machinePoolScope.ManagedMachinePool.Name)
 	}
 
+	if machinePoolScope.ManagedMachinePool.Spec.AWSLaunchTemplate != nil {
+		launchTemplateID := machinePoolScope.ManagedMachinePool.Status.LaunchTemplateID
+		launchTemplate, _, err := ec2Svc.GetLaunchTemplate(machinePoolScope.LaunchTemplateScope.Name())
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if launchTemplate == nil {
+			machinePoolScope.V(2).Info("Unable to locate launch template")
+			r.Recorder.Eventf(machinePoolScope.ManagedMachinePool, corev1.EventTypeNormal, "NoLaunchTemplateFound", "Unable to find matching launch template")
+			controllerutil.RemoveFinalizer(machinePoolScope.ManagedMachinePool, expinfrav1.ManagedMachinePoolFinalizer)
+			return ctrl.Result{}, nil
+		}
+
+		machinePoolScope.Info("deleting launch template", "name", launchTemplate.Name)
+		if err := ec2Svc.DeleteLaunchTemplate(*launchTemplateID); err != nil {
```

Comment:
Should the delete use the `launchTemplate.LaunchTemplateId` instead?

Reply:
There's no LaunchTemplateId field in `AWSLaunchTemplate` at this moment, that's why i'm using launchTemplateID instead. Please let me know if that makes sense. Thanks!